Assignment4/JSLWilfizz-999927/task07.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1a-xqsJzK0Oa1Z93ecEoYWyoKPQtan8YF
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+from rdflib.plugins.sparql import prepareQuery
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+# Visualize the results
+
+ns = Namespace("http://somewhere#")
+#With SPARQL
+q1 = prepareQuery('''
+  SELECT  ?sub  WHERE {
+    ?sub rdfs:subClassOf ns:LivingThing.
+  }
+  ''', initNs={"ns": ns}
+                  )
+for r in g.query(q1):
+    print(r.sub)
+print("--------------------------------")
+#With RDFlib
+
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    print(s)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+# Visualize the results
+
+ns = Namespace("http://somewhere#")
+#With SPARQL
+
+q1 = prepareQuery('''
+  SELECT  ?persona  WHERE {
+    ?persona rdf:type ns:Person
+  }
+  ''', initNs={"ns": ns}
+                  )
+for r in g.query(q1):
+    print(r.persona)
+
+q2 = prepareQuery('''
+  SELECT  ?person WHERE {
+    ?sub rdfs:subClassOf ns:Person.
+    ?person a ?sub.
+  }
+  ''', initNs={"ns": ns}
+                  )
```

Comment:
missing again recursive subclassof relationships, and the queries can be merge into one